scipy/stats/tests/test_fit.py

```diff
@@ -232,11 +278,14 @@ def test_fitstart(distname, shapes):
     assert dist._argcheck(*guess[:-2])
 
 
-def assert_nllf_less_or_close(dist, data, params1, params0, rtol=1e-7, atol=0):
-    nllf1 = dist.nnlf(params1, data)
-    nllf0 = dist.nnlf(params0, data)
-    if not (nllf1 < nllf0):
-        np.testing.assert_allclose(nllf1, nllf0, rtol=rtol, atol=atol)
+def assert_nlff_less_or_close(dist, data, params1, params0, rtol=1e-7, atol=0,
+                              nlff_name='nnlf'):
+    nlff = getattr(dist, nlff_name)
+    with np.errstate(invalid='ignore', divide='ignore'):
+        nlff1 = nlff(params1, data)
+        nlff0 = nlff(params0, data)
```

Comment:
`vonmises` warnings were because `vonmises` can produce a negative CDF. I don't think we need to worry about `vonmises`.

The rest were because of the issue described [above](https://github.com/scipy/scipy/pull/17141#discussion_r989710751).